indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1021 @@
+  private static final String COUNTER_GROUP = "Hadoop Druid Converter";
+  private static final String COUNTER_LOADED = "Loaded Bytes";
+  private static final String COUNTER_WRITTEN = "Written Bytes";
+  private static Injector injector = null;
+  private final HadoopDruidConverterConfig converterConfig;
+  private long loadedBytes = 0L;
+  private long writtenBytes = 0L;
+
+  public HadoopConverterJob(
+      HadoopDruidConverterConfig converterConfig
+  )
+  {
+    this.converterConfig = converterConfig;
+  }
+
+
+  public List<DataSegment> run() throws IOException
+  {
+    final JobConf jobConf = new JobConf();
+    jobConf.setKeepFailedTaskFiles(false);
+    for (Map.Entry<String, String> entry : converterConfig.getHadoopProperties().entrySet()) {
+      jobConf.set(entry.getKey(), entry.getValue(), "converterConfig.getHadoopProperties()");
+    }
+    final List<DataSegment> segments = converterConfig.getSegments();
+    if (segments.isEmpty()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "No segments found for datasource [%s]",
+              converterConfig.getDataSource()
+          )
+      );
+    }
+    converterConfigIntoConfiguration(converterConfig, segments, jobConf);
+    jobConf.setNumMapTasks(segments.size());
+    jobConf.setNumReduceTasks(0);
+    jobConf.setWorkingDirectory(new Path(converterConfig.getDistributedSuccessCache()));
+
+    setupSerializers(jobConf);
+
+    setJobName(jobConf, segments);
+
+    if (converterConfig.getJobPriority() != null) {
+      jobConf.setJobPriority(JobPriority.valueOf(converterConfig.getJobPriority()));
+    }
+
+    final Job job = Job.getInstance(jobConf);
+
+    job.setInputFormatClass(ConfigInputFormat.class);
+    job.setMapperClass(ConvertingMapper.class);
+    job.setMapOutputKeyClass(Text.class);
+    job.setMapOutputValueClass(Text.class);
+    job.setMapSpeculativeExecution(false);
+    job.setOutputFormatClass(ConvertingOutputFormat.class);
+
+    JobHelper.setupClasspath(JobHelper.distributedClassPath(jobConf.getWorkingDirectory()), job);
+
+    try {
+      job.submit();
+      log.info("Job %s submitted, status available at %s", job.getJobName(), job.getTrackingURL());
+      final boolean success = job.waitForCompletion(true);
+      if (!success) {
+        final TaskReport[] reports = job.getTaskReports(TaskType.MAP);
+        if (reports != null) {
+          for (final TaskReport report : reports) {
+            log.error("Error in task [%s] : %s", report.getTaskId(), Arrays.toString(report.getDiagnostics()));
+          }
+        }
+        cleanup(job);
+        return null;
+      }
+      try {
+        loadedBytes = job.getCounters().findCounter(COUNTER_GROUP, COUNTER_LOADED).getValue();
+        writtenBytes = job.getCounters().findCounter(COUNTER_GROUP, COUNTER_WRITTEN).getValue();
+      }
+      catch (IOException ex) {
+        log.error(ex, "Could not fetch counters");
```

Comment:
Probably not, if a JobHistory server is temporarily unavailable, it can encounter this error. Having each task emit alerts is probably overkill compared to a more general JobHistory checker.